tutorials/basic_1_square/README.md

```diff
@@ -0,0 +1,137 @@
+
+Flame's `Game` class is an implementation of a Game Loop, which the basic structure on which basically all games are built on. It is called Game Loop because it really works a an infinite loop that continues to iterate as long as the game is running. This loop can be divided on some steps:
+
+ - 1: Take input from the player
+ - 2: Process the logic for the next frame
+ - 3: Render the frame.
+
+On this tutorial we will focus the steps two and three.
+
+To process the logic of the game, we use the `update` method, which is ran always before the frame is rendered and receives a single argument, a `double` value called `dt` (delta time), which is the ammout os seconds from the current iteration and the last one. This delta time is very important so we can always know how to correct calculate movement, animations and etc.
+
+To render the frame, we use the `render` method, this method receives a single argument which is an instance of a `dart:ui` `Canvas` class. With that instance we can basically render anything we want. It is important to not that in this method, we should only contain render logic, any game logic should be put in the `update` method.
+
+Now that we have a better understanding of how the game structures works, lets start to plan our game.
+
+First things first, lets think on what variables and data structure we would need. For that, lets recap what we are building: A simple game where a square keeps bouncing forever from one side of the screen to the other. Thinking on this, we will need:
+
+ - A variable to tell us the speed of the square in logical pixels per second.
+ - A variable to inform which direction the square is moving.
+ - A structure to represent our square, which has a position and dimensions.
+
+With that in mind, check the example below, note the comments for explanations on which code section.
+
+
+```dart
+class MyGame extends Game {
+  // A constant speed, represented in logical pixels per second
+  static const int square_speed = 400;
+
+  // To represent our square we are using the Rect class from dart:ui
+  // which is a handle class to represent this type of data. We will be
+  // seeing other types of data classes in the future, but for this
+  // example, Rect will do fine for us.
+  late Rect squarePos;
+
+  // To reprent our direction, we will be using an int value, where 1 means
+  // going to the right, and -1 going to the left, this may seems like a too much
+  // simple way of representing a direction, and indeed it is, but this will
+  // will work fine for our small example and will make more sense when we implement
+  // the update method
+  int squareDirection = 1;
+
+  // The onLoad method is where all of the game initialization is supposed to go
+  // for this example, you may think that this square could just be initialized on the field
+  // declaration, and you are right, but for ludic reasons and to present the life cycle method
```

Comment:
Ludic reasons?

Reply:
I just looked on the dictionary and this word don't have the meaning that I though it did. I was going for something like "to teach you about this method".

Any suggestions here? Maybe "but for learning purposes"?